apps/studio/components/interfaces/Organization/BillingSettings/Subscription/DowngradeModal.tsx

```diff
@@ -39,7 +41,26 @@ const DowngradeModal = ({
   subscription,
   onClose,
   onConfirm,
+  projects,
 }: DowngradeModalProps) => {
+  // Filter out the micro addon as we're dealing with that separately
+  const previousProjectAddons =
+    subscription?.project_addons.flatMap((projectAddons) => {
+      const addons = projectAddons.addons.filter((it) => it.variant.identifier !== 'ci_micro')
+      if (!addons.length) {
+        return []
+      } else {
+        return {
+          ...projectAddons,
+          // Overwrite addons, filtered out the micro addon
+          addons,
+        }
+      }
+    }) || []
+
+  const hasInstancesOnMicro = projects.some((project) => project.infra_compute_size === 'micro')
+  const downgradingToNano = subscription?.nano_enabled === true
```

Comment:
What does this flag mean? Is `nano_enabled` on the current subscription or the subscription you're downgrading to?

Reply:
Also, tried to test this flow but `nano_enabled` doesn't exist yet on the Subscription API.

Reply:
This is forwards compatible, the flag will be exposed in an upcoming API PR - if the flag is missing, this will be false, which is correct

Reply:
With `nano_enabled` when downgrading from paid -> free and having micro instances, it will downgrade the instances to nano